src/Features/Core/Portable/Snippets/SnippetProviders/AbstractInlineStatementSnippetProvider.cs

```diff
@@ -65,7 +71,8 @@ protected sealed override async Task<TextChange> GenerateSnippetTextChangeAsync(
             var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
             _ = TryGetInlineExpression(targetToken, syntaxFacts, out var inlineExpression);
 
-            var statement = GenerateStatement(SyntaxGenerator.GetGenerator(document), syntaxContext, inlineExpression);
+            OriginalInlineExpression = inlineExpression;
+            var statement = GenerateStatement(SyntaxGenerator.GetGenerator(document), syntaxContext, inlineExpression?.WithoutLeadingTrivia());
```

Comment:
I believe I would prefer to continue passing `inlineExpression` here, remove the `OriginalInlineExpression` field, and update `AbstractIfSnippetProvider` to trim the leading trivia.

Reply:
The thing is that this is not only `if` snippet problem. For instance, here is `for` snippet without PR changes:
![devenv_NMS9BpivkY](https://github.com/dotnet/roslyn/assets/70431552/ac6dfcfa-01d5-4133-af5f-18ee41331212)
So my choise was between relying on each individual snippet to strip trivia and having this hack with `OriginalInlineExpression` field. Since only fraction of snippets needs semantic information about the node, the latter solution seems more preferable. Though I would appreciate if you come up with solution, that won't envolve any sort of hacks like this

Reply:
The parameters in such a case could be `inlineExpression`, `inlineExpressionWithoutTrivia`

Reply:
This isn't much different from relying on each snippet to do the job on its own. We are still delegating decision to them, but in a bit more explicit form. And since each snippet can get expression without trivia itself, it seems kinda pointless to pass 2 variants of the same expression as parameters.

Reply:
Yes, the explicitness reduces the chance of accidental misuse of values. The important part for me is the ability of an implementation of `GenerateStatement` to access the semantic model based on a parameter.

Reply:
Well, I managed to find a way, that isn't as hacky as what I did before. We just manually walk statement's descendant nodes and find equvivalent to `inlineExpression`. Then we replace it in the statement with itself without leading trivia. I left comments in code explaining why we do that. I don't think such kinda unclear logic is ok for public API, but as long as it is internal, I guess, it is acceptable to leave it like this

Reply:
I've run into a few cases where we did things like this, and it makes future debugging very difficult (a feature is returning a specific value, only to find that the specific value isn't used in its original form but some other part is manipulating it). I would really prefer for each feature to own its behavior for manipulating trivia. It involves changes to a couple extra places, but historically has proven to be more robust.

Reply:
Ok, your argument is convinving enough, I give up on centralizing this thing. Let's just put this logic in every snippet and call it a day. Now need to remember to implement and test such trivia cases in every future inline statement snippet, this is unfortunate(